compact_str/src/repr/mod.rs

```diff
@@ -165,9 +166,14 @@ impl Repr {
             // We only hit this case if the provided String is > 16MB and we're on a 32-bit arch. We
             // expect it to be unlikely, thus we hint that to the compiler
             capacity_on_heap(s)
-        } else if og_cap == 0 {
+        } else if s.is_empty() {
```

Comment:
If `should_inline` is `false` and `s` contains a large pre-allocated buffer, then should we re-use the buffer to avoid alloc?

Reply:
Yeah good thought, I updated the PR to be based on capacity now instead of length

Reason being, when created a `String` will only allocate as much space as needed. When appending to a string though it grows at a rate of 2x to amortize the cost of re-allocations. This leaves us with 2 scenarios:

1. capacity == length, in which case it doesn't matter which we check
2. capacity > length, this indicates the user appended to the string, or specifically allocated more space, which it seems like we'd want to keep

Reply:
hmmm, actually it seems like `Clone`-ing today actually doesn't preserve capacity, [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5a9d2bbfccfcd5f53f027cd3166d5ee0), so my assumption that users would want to preserve capacity might be wrong, let me think more